src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -354,9 +357,9 @@ protected NestedResourceInfoScope ParentNestedResourceInfoScope
 
                     // Get the resourceSet's parent (if any)

```

Comment:
>if any [](start = 53, length = 6)

the comment suggests there may not be a parent, but we never check whether it is null. seems like we should change the comment or the code.